src/Aspire.Hosting.Azure/AzureBicepResource.cs

```diff
@@ -47,6 +47,11 @@ public class AzureBicepResource(string name, string? templateFile = null, string
     /// </summary>
     public TaskCompletionSource? ProvisioningTaskCompletionSource { get; set; }
 
+    /// <summary>
+    /// The scope of the resource.
+    /// </summary>
+    public Dictionary<string, object> Scope { get; } = [];
```

Comment:
Can we document this a little better? I'm not sure what the key nor value of this dictionary represents from `The scope of the resource.`.

Reply:
Mostly, I'm wondering:

1. why there would ever be more than 1 (this is a Dictionary)
2. What is the key to this dictionary used for? Are there well-known keys?

Reply:
I've updated the comment here with a remark that contains more details. I'll answer the questions here for posterities sake.

1. It's a dictionary to support adding support for defining resources in separate resource groups _and_ subscriptions in the future.
2. The `resourceGroup` is the only well-known key here. AZD will disregard everything else.